tests/test_server_basic.py

```diff
@@ -718,6 +718,165 @@ def test_list_file(dispatcher_live_fixture):
     assert job_id == calculated_job_id
 
 
+def test_catalog_file(dispatcher_live_fixture):
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+        "roles": "unige-hpc-full, general",
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    params = {
+        **default_params,
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token
+    }
+
+    file_path = DispatcherJobState.create_catalog_file(catalog_value=5)
+
+    list_file = open(file_path)
+
+    jdata = ask(server,
+                params,
+                expected_query_status=["done"],
+                max_time_s=150,
+                method='post',
+                files={"user_catalog_file": list_file.read()}
+                )
+
+    list_file.close()
+    assert 'user_catalog_file' in jdata['products']['analysis_parameters']
+    # test job_id
+    job_id = jdata['products']['job_id']
+    session_id = jdata['session_id']
+    # adapting some values to string
+    for k, v in params.items():
+        params[k] = str(v)
+
+    restricted_par_dic = InstrumentQueryBackEnd.restricted_par_dic({**params, "user_catalog_file": f'temp_sid_{session_id}/user_catalog_file', "sub": "mtm@mtmco.net"})
+    calculated_job_id = make_hash(restricted_par_dic)
+
+    assert job_id == calculated_job_id
+
+
+def test_user_catalog(dispatcher_live_fixture):
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+        "roles": "unige-hpc-full, general",
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    selected_catalog_dict = dict(
```

Comment:
just FYI, this is not exactly how we would create these catalogs, since it's a bit verbose this way. 
see examples in this [repository](https://github.com/oda-hub/oda_api_benchmark/blob/master/examples/V404_Cyg_analysis.ipynb)

however, directly constructing as you did has an advantage of expressing the format in here.

Reply:
thanks

A good improvement for the future is to have a dedicated fixture that would build a catalog in the most appropriate way

Reply:
maybe at some point it would be even useful  for users to construct the catalog more intuitively.